### PR TITLE
 12 add-search-functionality

### LINKED ISSUE #12: Add search functionality
## Description
Please create search command that will filter all contacts by checking if entered text is a sub-string of str representation of `Record`

* Add command to main.py
* Extend `AddressBook` class with a new method `find`

### PR DESCRIPTION
**Pull request by NotJod563**
Function was called search instead of find because we already got find :D

### CODE REVIEW

File: address_book/address_book.py [hunk 1]
```diff
@@ -76,14 +76,36 @@ def birthdays(self, args):
...
+            phones_str = [str(phone).lower().strip() for phone in record.phones]
+            email_str = str(record.email).lower().strip() if record.email else ""
+
+            if (search_lowered in name_str or
+                any(search_lowered in phone for phone in phones_str) or
+                (record.email and search_lowered in email_str)):
+                search_results.append(record)
+
+        if search_results:
+            return "\n".join([str(record) for record in search_results])
```
Comment (byymster) [inline address_book/address_book.py]:
```suggestion
            return search_results
```
you can use pretty_print from utils.py in main.py to do the same

File: address_book/address_book.py [hunk 2]
```diff
@@ -76,14 +76,36 @@ def birthdays(self, args):
...
+        search_term = " ".join(search_term) if isinstance(search_term, list) else search_term
+        search_results = []
+        search_lowered = search_term.lower().strip()
+
+        for record in self.data.values():
+            name_str = str(record.name.value).lower().strip()
+            phones_str = [str(phone).lower().strip() for phone in record.phones]
+            email_str = str(record.email).lower().strip() if record.email else ""
+
+            if (search_lowered in name_str or
```
Comment (byymster) [inline address_book/address_book.py]: @NotJod563 Maybe it's better to concatenate all strings (data) and search in them to reduce amount of conditions in if statement?

File: main.py [hunk 3]
```diff
@@ -25,6 +25,8 @@ def main():
                     print(contacts.find(args[0]))
                 case 'all':
                     print(contacts.all())
+                case 'search':
+                    print(contacts.search(args))
```
Comment (byymster) [inline main.py]:
```suggestion
                    pretty_print(contacts.search(args))
```

**Review by byymster — CHANGES_REQUESTED** [review]

File: address_book/address_book.py (hunk 1 above)
Comment (NotJod563) [reply address_book/address_book.py]: Updated in second commit

File: address_book/address_book.py (hunk 2 above)
Comment (NotJod563) [reply address_book/address_book.py]: Updated in second commit

File: main.py (hunk 3 above)
Comment (NotJod563) [reply main.py]: Updated in second commit

**Review by byymster — DISMISSED** [review]
LGTM

**Review by byymster — DISMISSED** [review]

**Review by byymster — APPROVED** [review]

**Review by MAlexAT — APPROVED** [review]

**Pull request merged**